lib/absinthe/phase/document/validation/variables_are_input_types.ex

```diff
@@ -20,19 +20,22 @@ defmodule Absinthe.Phase.Document.Validation.VariablesAreInputTypes do
 
   # Find variable definitions
   @spec handle_node(Blueprint.node_t(), Schema.t()) :: Blueprint.node_t()
-  defp handle_node(%Blueprint.Document.VariableDefinition{schema_node: nil} = node, _) do
```

Comment:
The problem with this is that `schema_node` can be "wrapped" like `%Absinthe.Type.NonNull{of_type: nil}` so it doesn't actually catch all the cases when there is an un-known type

Reply:
Since, as you say, a `nil` `schema_node` can be "wrapped," it seems to make sense to update the `@spec` for `Type.unwrap/1` to add `nil` as a possible return value.

I've added a commit to that end (passes locally).